src/app/types.py

```diff
@@ -18,6 +18,11 @@ def __call__(
         ...
 
 
+class ExistCheckAssertion(Protocol):
+    def __call__(self, model: type[Model], **filter: Any) -> None:
+        ...
+
+
```

Comment:
Generic would be more flexible here:


```suggestion
Model = TypeVar('Model')

class ExistCheckAssertion(Generic[Model]):
    def __call__(self, model: type[Model], **filter: Any) -> None:
        ...

```

Usage:

```python
def test_blabla(assert_exists: ExistCheckAssertion[<ActualModelHere>]): ...
```

Reply:
I think it needs to be done after this [PR](https://github.com/Yakov-Varnaev/bbq/pull/43). We change the same files.